TerriaJS/terriajs
# Restyle help panel

### What this PR does

Fixes #5636 #4670 

Changes:
- Re-styles help panel menu items based on the new design from @philipgrimmett 
- Adds map user guide link to mobile menu if it is supplied in `Terria.configParameters.helpItems`.
- Fixes app crash when loading in mobile Safari browser. This is happening because of a render loop in `TooltipWrapperRaw` component as the following check fails. Even though help panel is not shown anywhere in mobile, the help panel is rendered eagerly. I assume it was done to animate opening of help panel. I have changed it so that the `HelpPanel` component is rendered only when activated (https://github.com/TerriaJS/terriajs/pull/5807/commits/daea39115c015108ce9deaeafacd0c758601347e). I have also made changes to retain the animation when help panel is opened (https://github.com/TerriaJS/terriajs/pull/5807/commits/2195fc73411b27e933a8c908e2558a4899ab3e35).

https://github.com/TerriaJS/terriajs/blob/3c090fa5bccbe88c030f17b317b839a05080271d/lib/ReactViews/Generic/TooltipWrapper.tsx#L106-L111

On mobile Safari the above check passes each time (difference is above 5 maybe because of high pixel density?) and an infinite update loop is set up crashing the app. Fixing the root cause will require a deeper surgery. My fix above is still a good fix because we shouldn't be rendering help panel in mobile. I have made a ticket capturing the problem https://github.com/TerriaJS/terriajs/issues/5827

### Testing

* Open [this link](http://ci.terria.io/restyle-help-panel/#configUrl=https://gist.githubusercontent.com/na9da/b326147a3cc7a066d6501ecef2b93960/raw/834af0d311ccf1b4d6b73872b0b3cfdb6a914412/config-with-map-user-guide.json)
* Check out the help panel. You should see a link to the map user guide.
* Now switch to mobile view from the developer console
* Open the hamburger menu
* You should see a link to the map user guide.

### Checklist
